homeassistant/components/device_tracker/__init__.py

```diff
@@ -95,30 +103,43 @@ def setup(hass, config):
     """Setup device tracker."""
     yaml_path = hass.config.path(YAML_DEVICES)
 
-    conf = config.get(DOMAIN, {})
+    conf = CONFIG_SCHEMA(config).get(DOMAIN, [])
 
-    # Config can be an empty list. In that case, substitute a dict
-    if isinstance(conf, list):
-        conf = conf[0] if len(conf) > 0 else {}
+    def _get_conf_value(key_name, default):
```

Comment:
With this approach you can add it anywhere (first, last, or more than one places), but it will warn you if you use different values (we can and will only honor the first one)

It will solve issues, where you add a dev tracker in the future, but inserting it first in the list for example. Currently it will just silently ignore